libs/sdk-core/src/input_parser.rs

```diff
@@ -261,17 +261,26 @@ pub(crate) async fn post_and_log_response(url: &str, body: Option<String>) -> Sd
 pub(crate) async fn get_and_log_response(url: &str) -> SdkResult<String> {
     debug!("Making GET request to: {url}");
 
-    let raw_body = get_reqwest_client()?
+    let response = get_reqwest_client()?
         .get(url)
         .send()
         .await
-        .map_err(|e| SdkError::ServiceConnectivity { err: e.to_string() })?
-        .text()
-        .await
         .map_err(|e| SdkError::ServiceConnectivity { err: e.to_string() })?;
-    debug!("Received raw response body: {raw_body}");
-
-    Ok(raw_body)
+    match response.status().is_success() {
+        true => response
+            .text()
+            .await
+            .map_err(|e| SdkError::ServiceConnectivity { err: e.to_string() }),
+        false => {
```

Comment:
Unfortunately this would break error handling for LNURL.

Some specs (like LNURL but maybe others too) define errors in terms of how the response payload looks like, regardless of HTTP status.

If we now throw `Err` here in the `GET` response parsing in case of HTTP error code, this means we will not return anymore the LNURL error JSON payload

```json
{"status": "ERROR", "reason": "error details..."}
```

which means we lose the info on the error cause indicated by the server (not enough funds, etc).

IMO we need to return both the HTTP status and the raw payload, then let the caller decide. For mempool API calls, HTTP error code means error. For LNURL, the HTTP error code is only one mode of failure, the other is the error payload (can be returned with HTTP 200 or HTTP error).

Reply:
I tested it and indeed:

with the PR, trying to LNURL-withdraw from a wallet with no funds returns

> Error: Service connectivity: Service connectivity: GET request https://stacker.news/api/lnwith?k1=ce529c69344c7ae... failed with status: 400 Bad Request

whereas on `main` the same LNURL call returns

```json
{
  "ErrorStatus": {
    "data": {
      "reason": "insufficient funds"
    }
  }
}
```

These LNURL error objects are useful to apps integrating with us, as they can surface the error reason to the end-user.

Reply:
How about #990? IMO it covers both cases.

Reply:
I think it would be good to add test cases for the two ways a response is handled

Reply:
HTTP status has (should have) no meaning at all in LNURL: https://github.com/lnurl/luds/blob/luds/01.md#http-status-codes-and-content-type

Reply:
@dangeross added test. Now both cases are tested.